File: No.2490-Bees-master/api/src/main/java/org/bees/biz/service/PackagesService.java
package org.bees.biz.service;

import java.io.IOException;
import java.lang.reflect.InvocationTargetException;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import org.apache.commons.beanutils.PropertyUtils;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.bees.biz.persistence.manager.PackagesManager;
import org.bees.biz.persistence.manager.PackagesOrderManager;
import org.bees.biz.persistence.manager.PackagesPriceManager;
import org.bees.biz.persistence.manager.UserManager;
import org.bees.biz.persistence.model.Packages;
import org.bees.biz.persistence.model.PackagesOrder;
import org.bees.biz.persistence.model.PackagesPrice;
import org.bees.biz.persistence.model.User;
import org.bees.biz.service.dto.IndexCommentDto;
import org.bees.biz.service.dto.PackagesCommentDto;
import org.bees.utils.Const;
import org.codehaus.jackson.JsonParseException;
import org.codehaus.jackson.map.JsonMappingException;
import org.codehaus.jackson.map.ObjectMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import projects.commons.exception.ServiceException;
import projects.commons.utils.ValidateUtils;

@Service
public class PackagesService {
	
	@Autowired
	private PackagesManager packagesManager;
	
	@Autowired
	private PackagesPriceManager packagesPriceManager;
	
	@Autowired
	private PackagesOrderManager packagesOrderManager;
	
	@Autowired
	private UserManager userManager;
	
	

	@SuppressWarnings("unused")
	private final static Log LOG = LogFactory.getLog(PackagesService.class);

	public Packages getInfoById(Integer id) {		
		return this.packagesManager.getInfoById(id);
	}

	public List<PackagesPrice> getPriceByPid(Integer id) {
		return this.packagesPriceManager.getPriceByPid(id);
	}

	public List<PackagesCommentDto> getCommentsByPid(Integer pid) {
		List<PackagesOrder> orders=this.packagesOrderManager.getOrders(0,null,null,null,null,null,pid,null,null);
		List<PackagesCommentDto> dtos=new LinkedList<PackagesCommentDto>();
		for (PackagesOrder order : orders) {
			PackagesCommentDto dto = new PackagesCommentDto();
			try {
				PropertyUtils.copyProperties(dto, order);
				User user=this.userManager.getInfoById(order.getUid());
				dto.setName(user.getNickname());
				dto.setAvatar(user.getAvatar());
				
			} catch (IllegalAccessException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			} catch (InvocationTargetException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			} catch (NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}
		return dtos;
	}

	public List<IndexCommentDto> getCommentList(int start, int limit, Integer order) {
		List<PackagesOrder> orders=this.packagesOrderManager.getOrders(null,null,null,Const.ORDER_STATUS_FOUR,null,null,null,start,limit);
		List<IndexCommentDto> dtos=new LinkedList<IndexCommentDto>();
		for (PackagesOrder tempOrder : orders) {
			IndexCommentDto dto = new IndexCommentDto();
			try {
				PropertyUtils.copyProperties(dto, tempOrder);
				User user=this.userManager.getInfoById(tempOrder.getUid());
				dto.setName(user.getNickname());
				dto.setImgUrl(user.getAvatar());
				dto.setRoomNo(user.getLiveRoom());
				dto.setId(tempOrder.getId());
				dto.setDesc(user.getDescription());
				dto.setIntro(tempOrder.getComment());
				dto.setUid(user.getId());
				String[] names=tempOrder.getPackageName().split("/");
				dto.setType(names[names.length-1]);
				
			} catch (IllegalAccessException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			} catch (InvocationTargetException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			} catch (NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}
		return dtos;
	}

	public List<Packages> getPackagesList(int start, int limit, Integer order) {
		return this.packagesManager.getList(start,limit);
	}

	@SuppressWarnings("unchecked")
	public boolean add(String name, String content, String titleOne,
			String titleTwo, String titleThree, String pic, String prices) {
		int id=this.packagesManager.add(name,content,titleOne,titleTwo,titleThree,pic);
		ObjectMapper objectMapper = new ObjectMapper();
		List<Map<String, String>> priecLists = null;
		try {
			priecLists = objectMapper.readValue(prices, List.class);
			for (Map<String, String> map : priecLists) {
				this.packagesPriceManager.add(id,map.get("name"),map.get("price"));
			}		
		} catch (JsonParseException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (JsonMappingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}		
		return true;
	}

	public boolean del(Integer id) {
		return this.packagesManager.del(id);
	}

	@SuppressWarnings("unchecked")
	public boolean edit(Integer id, String name, String content,
			String titleOne, String titleTwo, String titleThree, String pic, String prices) {
		this.packagesManager.edit(id,name,content,titleOne,titleTwo,titleThree,pic);
		List<PackagesPrice> oldPrices=this.packagesPriceManager.getPriceByPid(id);
		List<Integer> ids = new LinkedList<Integer>();
		if(!ValidateUtils.isNull(oldPrices)){
			for (PackagesPrice packagesPrice : oldPrices) {
				ids.add(packagesPrice.getId());
			}
		}
		ObjectMapper objectMapper = new ObjectMapper();
		List<Map<String, String>> priecLists = null;
		try {
			priecLists = objectMapper.readValue(prices, List.class);
			for (Map<String, String> map : priecLists) {
				if(ValidateUtils.isNull(map.get("id"))){
					this.packagesPriceManager.add(id,map.get("name"),map.get("price"));
				}else{
					this.packagesPriceManager.edit(Integer.valueOf(map.get("id")),map.get("name"),map.get("price"));
					ids.remove(ids.indexOf(Integer.valueOf(map.get("id"))));
				}
			}
			if(!ids.isEmpty()){
				this.packagesPriceManager.dels(ids);
			}
		} catch (JsonParseException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (JsonMappingException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}		
		return false;
	}

	public PackagesPrice getPriceByid(Integer id) { 
		return this.packagesPriceManager.getInfoById(id);
	}

	public int getCommentCount(Integer pid) {

		return this.packagesOrderManager.getCommentCount(pid);
	}

	public boolean delPrice(Integer id) {
		List<Integer> ids=new LinkedList<Integer>();
		ids.add(id);
		this.packagesPriceManager.dels(ids);
		return true;
	}

	public boolean addComment(Integer userId, Long orderId, String comment) {
		PackagesOrder order=this.packagesOrderManager.getInfoById(orderId);
		if(order.getUid().intValue()!=userId.intValue()){
			throw new ServiceException("评论失效");
		}else{
			boolean res=this.packagesOrderManager.addComment(orderId,comment);
			return res;
		}
		
	}
	
	


}

File: beesInterface/bees_package/packagesService.py
#coding:utf-8
from models import *
import json
import sys
from utils.const import Const
from bees_package.models import Packages, PackagesOrder, PackagesPrice
from exception.serviceException import ServiceException
from bees_user.userService import UserService
from utils.validateUtils import ValidateUtils
import django.utils.timezone as timezone

sys.path.append('..')

class PackagesService:
	@staticmethod
	def getInfoById(id):
		try:
			return Packages.objects.get(id=id, del_flag=Const.DEL_FLAG_NOT_DEL)
		except:
			return None

	@staticmethod
	def getPriceByPid(id):
		packagesPrices = PackagesPrice.objects.filter(package_id=id, del_flag=Const.DEL_FLAG_NOT_DEL)
		dtos = []
		for packagesPrice in packagesPrices:
			dtos.append(packagesPrice.getDict())
		return dtos

	@staticmethod
	def getCommentsByPid(pid):
		orders = PackagesOrder.objects.filter(id=0, package_id=pid)
		dtos = []
		for order in orders:
			dto = order.getOrderDto()
			user = UserService.getInfoById(order.uid)
			dto['name'] = user.nickname
			dto['avatar'] = user.avatar
			dtos.append(dto)
		return dtos

	@staticmethod
	def getCommentList(start, limit, order): # order
		orders = PackagesOrder.objects.filter(status=4)
		if order:
			orders = orders.order_by(order)
		if start:
			pass
		else:
			start = 0
		if limit:
			orders = orders[start: start+limit]
		else:
			orders = orders[start:]
		dtos = []
		for tempOrder in orders:
			dto = tempOrder.getOrderDto()
			user = UserService.getInfoById(tempOrder.uid)
			dto['name'] = user.nickname
			dto['imgUrl'] = user.avatar
			dto['roomNo'] = user.liveRoom
			dto['desc'] = user.description
			dto['uid'] = user.uid
			dto['type'] = PackagesOrder.__name__
			dtos.append(dto)
		return dtos

	@staticmethod
	def getPackagesList(start, limit, order):
		if ValidateUtils.isNull(start):
			start = 0
		return ['未实现']
		# if ValidateUtils.isNull(limit):
		# 	return Packages.objects.all()[start:]
		# else:
		# 	return Packages.objects.all()[start: start+limit]
		return dtos

	@staticmethod
	def  add(name, content, titleOne,
			titleTwo, titleThree, pic, prices):
		packages = Packages()
		packages.name = name
		packages.content = content
		packages.titleOne = titleOne
		packages.titleTwo = titleTwo
		packages.titleThree = titleThree
		packages.pic = pic
		packages.save()
		id = packages.id
		priecLists = json.loads(prices)
		for map in priecLists:
			try:
				packagesPrice = PackagesPrice(package_id=id, name=map.name, price=map.price)
				packagesPrice.save()
			except:
				return False
		return True

	@staticmethod
	def delete(id):
		try:
			packages = Packages.objects.get(id=id)
			packages.delete()
		except:
			return False
		return True

	@staticmethod
	def edit(id, name, content, titleOne, titleTwo, titleThree, pic, prices):
		try:
			packages = Packages.objects.get(id=id)
			# packages.add_time = datetime.now()
			packages.name = name
			packages.content = content
			packages.title_one = titleOne
			packages.title_two = titleTwo
			packages.title_three = titleThree
			packages.pic = pic
			packages.save()
		except Exception, e:
			return False
		oldPrices = PackagesPrice.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL, package_id=id)
		ids = []
		if not ValidateUtils.isNull(oldPrices):
			for packagesPrice in oldPrices:
				ids.append(packagesPrice.id)
		# print 'prices:', prices
		priecLists = json.loads(prices)
		for map in priecLists:
			if ValidateUtils.isNull(map.get('id')):
				packagesPrice = PackagesPrice()
				packagesPrice.package_id = id
				# packagesPrice.add_time = datetime.datetime.now()
				packagesPrice.name = map.get('name')
				packagesPrice.price = map.get('price')
				packagesPrice.save()
			else:
				packagesPrice = PackagesPrice.objects.get(id=map.get('id'))
				packagesPrice.name = map.get('name')
				packagesPrice.price = map.get('price')
				packagesPrice.save()
				ids.remove(int(map.get('id')))
		if len(ids) > 0:
			PackagesPrice.objects.filter(id__in=ids).update(del_flag=Const.DEL_FLAG_DEL)
		return True

	@staticmethod
	def getPriceByid(id):
		try:
			return PackagesPrice.objects.get(id=id)
		except:
			return None

	@staticmethod
	def getCommentCount(pid):
		packagesOrders = PackagesOrder.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL)
		if not ValidateUtils.isNull(pid):
			packagesOrders = packagesOrders.filter(package_id=pid)
		return packagesOrders.count()

	@staticmethod
	def delPrice(id):
		try:
			packagesPrice = PackagesPrice.objects.get(id=id)
			packagesPrice.del_flag = Const.DEL_FLAG_DEL
			packagesPrice.save()
			return True
		except:
			return False

	@staticmethod
	def addComment(userId, orderId, comment):
		try:
			order = PackagesOrder.objects.get(id=orderId)
			if order.uid != userId:
				raise ServiceException("评论失效")
			else:
				order.status = Const.ORDER_STATUS_FOUR
				order.comment = comment
				order.comment_time = timezone.now()
				order.save()
				return True
		except:
			return False
